Hide internalName from buyer

lesser1 opened the pull request:
Changes:
    - Add hint to text field of Add Product dialog.
    - Remove internal name from product page (ProductView).

[Gropius](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/6af75f49-6f87-40a1-b905-ebbc8a100725)

### Definition of Done
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented

lesser1 committed:
Hide internalName from buyer

    - Add hint to text field of Add Product dialog.
    - Remove internal name from product page (ProductView).

nk-coding reviewed (changes requested):
while it does not display it, it still fetches it which still causes an error

lesser1 committed:
Add extra product query for buyer

- Introduce new GraphQL query to get product information only relevant
  to the buyer.
- Modify ProductView.vue: Differentiate between two product queries:
    - one for the buyer
    - one for admins and employees
- Add extra error message for new GraphQL query

lesser1 commented:
> while it does not display it, it still fetches it which still causes an error

1296c8b1f6be4c4679ee829baeeca0d4310dc805 -- changes

d057d1cbe960a5cbe799e9556ad709bec36c5012 -- merge due to conflicts with main

lesser1 committed:
Merge branch 'main' into feature/hide-internal-name-from-buyer

lesser1 committed:
Run 'yarn format' to fix prettier issues
